### PR TITLE
Remove Goptions.opt_name field

### PR DESCRIPTION
**Pull request by SkySkimmer**
The standard use is to repeat the option keywords in lowercase, which
is basically useless.

En passant add doc entry for Dump Arith.

Overlays: https://github.com/coq-community/paramcoq/pull/46 https://github.com/unicoq/unicoq/pull/34 https://github.com/mattam82/Coq-Equations/pull/259 https://github.com/lukaszcz/coqhammer/pull/55

### CODE REVIEW

**Comment by SkySkimmer** [comment]
We probably don't need quite so many reviewers ;)

**Review by ppedrot — APPROVED** [review]
Fine.

**Comment by ppedrot** [comment]
You need a lot of overlays though...

**Comment by SkySkimmer** [comment]
>You need a lot of overlays though...

It's not that bad.

**Comment by SkySkimmer** [comment]
Are we merging this before overlays get out of date?

**Comment by ppedrot** [comment]
OK, will merge when CI passes.

**Comment by ppedrot** [comment]
Suspicious CI failures, did you rebase the overlays?

**Comment by SkySkimmer** [comment]
There's no overlay for quickchick

**Comment by ppedrot** [comment]
Right, that's unrelated overlay desynchronization. Will disregard the failure then.

File: doc/sphinx/addendum/micromega.rst [hunk 1]
```diff
@@ -35,6 +35,11 @@ tactics for solving arithmetic goals over :math:`\mathbb{Q}`,
    use the Simplex method for solving linear goals. If it is not set,
    the decision procedures are using Fourier elimination.
 
+.. flag:: Dump Arith
```
Comment (Zimmi48) [inline doc/sphinx/addendum/micromega.rst]: This is an option, not a flag.

**Review by Zimmi48 — CHANGES_REQUESTED** [review]

**Review by Zimmi48 — APPROVED** [review]

**Pull request merged**